tests/unit/single_table/test_base.py

```diff
@@ -104,3 +108,584 @@ def test_fit(self):
+            'name': ['John', 'Doe', 'John Doe'],
+            'salary': [60., 70., 80.]
+        })
+        second_data = pd.DataFrame({
+            'name': ['Johana', 'Doe', 'Johana Doe'],
+            'salary': [65., 75., 85.]
+        })
+        instance = Mock()
+        instance._sample_batch.side_effect = [first_data, second_data]
+
+        # Run
+        result = BaseSynthesizer._sample_in_batches(
+            instance,
+            num_rows=6,
+            batch_size=3,
+            max_tries_per_batch=100,
+            conditions='conditions',
+            transformed_conditions='transformed_conditions',
+            float_rtol=0.02,
+            progress_bar='progress_bar',
+            output_file_path='output_file_path',
+        )
+
+        # Assert
+        expected_result = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe', 'Johana', 'Doe', 'Johana Doe'],
+            'salary': [60., 70., 80., 65., 75., 85.]
+        })
+        pd.testing.assert_frame_equal(result, expected_result)
+        assert instance._sample_batch.call_count == 2
+        expected_call = call(
+            batch_size=3,
+            max_tries=100,
+            conditions='conditions',
+            transformed_conditions='transformed_conditions',
+            float_rtol=0.02,
+            progress_bar='progress_bar',
+            output_file_path='output_file_path',
+        )
+        assert expected_call == instance._sample_batch.call_args_list[0]
+        assert expected_call == instance._sample_batch.call_args_list[1]
+
+    def test__conditionally_sample_rows(self):
+        """Test when sampled rows is bigger than 0."""
+        # Setup
+        transformed_data = pd.DataFrame({
+            COND_IDX: [0, 1, 2],
+            'salary.value': [65., 75., 85.]
+        })
+        data = pd.DataFrame({
+            'name': ['Johana', 'Doe', 'Johana Doe'],
+            'salary': [65., 75., 85.],
+        })
+        instance = Mock()
+        instance._sample_in_batches.return_value = data
+        condition = Mock()
+        transformed_condition = Mock()
+
+        # Run
+        result = BaseSynthesizer._conditionally_sample_rows(
+            instance,
+            transformed_data,
+            condition,
+            transformed_condition,
+        )
+
+        # Assert
+        expected_data = pd.DataFrame({
+            'name': ['Johana', 'Doe', 'Johana Doe'],
+            'salary': [65., 75., 85.],
+            COND_IDX: [0, 1, 2]
+        })
+        pd.testing.assert_frame_equal(expected_data, result)
+        instance._sample_in_batches.assert_called_once_with(
+            num_rows=3,
+            batch_size=3,
+            max_tries_per_batch=None,
+            conditions=condition,
+            transformed_conditions=transformed_condition,
+            float_rtol=0.01,
+            progress_bar=None,
+            output_file_path=None
+        )
+
+    def test__conditionally_sample_rows_raises_value_error(self):
```

Comment:
I think the name here could describe why the error is raise, eg. `test__conditionally_sample_rows_no_rows_sampled_error`